plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3268,12 +3315,19 @@ public void testOptimize()
                 .collect(toImmutableList());
     }
 
-    private List<String> getAllDataFilesFromTableDirectory(String tableName)
+    protected List<String> getAllDataFilesFromTableDirectory(String tableName)
             throws IOException
     {
         String schema = getSession().getSchema().orElseThrow();
-        Path tableDataDir = getDistributedQueryRunner().getCoordinator().getBaseDataDir().resolve("iceberg_data").resolve(schema).resolve(tableName).resolve("data");
-        try (Stream<Path> walk = Files.walk(tableDataDir)) {
+        String tableDataDir = getDistributedQueryRunner().getCoordinator()
+                .getBaseDataDir()
+                .resolve("iceberg_data")
+                .resolve(schema)
+                .resolve(tableName)
+                .resolve("data")
+                .toAbsolutePath()
+                .toString();
```

Comment:
pls remove `.toString()` method call. In the line below you make the conversion to `Path` again.